test/widgets/autocomplete_test.dart

```diff
@@ -372,6 +372,56 @@ void main() {
       expected ? displaySubject.findsOne(): displaySubject.findsNothing();
     }
 
+    testWidgets('text emoji autocomplete item must have correct height', (tester) async {
+      // Regression test for #1587
+      tester.platformDispatcher.textScaleFactorTestValue = 2.0;
+      addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
+
+      final composeInputFinder = await setupToComposeInput(tester);
+
+      // Set up a realm emoji so we have a reliable image emoji to test against.
+      await store.handleEvent(RealmEmojiUpdateEvent(id: 1, realmEmoji: {
+        '1': eg.realmEmojiItem(emojiCode: '1', emojiName: 'an_image_emoji'),
+      }));
+      await tester.pumpAndSettle(); // Ensure store update is fully processed.
+
+      // Phase 1: Get the height of the item when it has an image.
+      // This two-step `enterText` call is a more reliable way to trigger the autocomplete view.
```

Comment:
```suggestion
      // TODO(#226): Remove this extra edit when this bug is fixed.
```

(here and the other place(s) where it applies)